### ISSUE TITLE
Press enter to dial

### ISSUE THREAD

**Issue by pvint** [BelledonneCommunications/linphone-desktop#169]
Feature request: After entering a number it would be very nice to simply have the number called when you press enter

**Comment by pvint** [comment]
Ah, glad you pointed out the "so long as you have @address" - I was sure that it works sometimes, and I think that explains it.

I haven't looked at the code for this yet but I will do so if time allows. Let me know if you make any progress or if I can help.

**Comment by adamcjohnson95** [comment]
I finally figured it out.

It takes a few adjustments to multiple files if you want to give the user more freedom to type what they want. Listed below are the changes I made, if anyone sees an issue with what I did let me (and more importantly anyone who may try and copy this behavior) know.

------------------------------------------------------------------------

`ui\modules\LinPhone\Form\SearchBox.qml`
`ui\modules\Common\Form\SearchBox.qml`

1. comment out the line `onTextChanged: _filter(text)`

Extra: I also added  `Keys.onEnterPressed: { searchBox.enterPressed() searchBox.closeMenu() }` after the `onReturnPressed:` handlers just in case. Not really sure if this helps with anything, as I'm new to QT, but I figured some redundancy wouldn't hurt.


------------------------------------------------------------------------

`\submodules\linphone\coreapi\proxy.c`

1. Changed the code in the function `linphone_proxy_config_normalize_sip_uri`. Basically just added a couple if() statements in. If it doesn't start with `sip:`, prepend it to the username that was passed. Then if it didn't the `@` symbol isn't in the name, append the `@address` that I wanted to be default. 

This means that basically any number can try and initiate a call. I set it up like this because the switch that will be handling the calls will determine what are valid numbers. Allows for extension dialing and international numbers to be used. 

**Attention:**
**Contact the creators before duplicating this behavior. There may be issues with this when using default/Linphone addresses and services. I'm working on this with a company that has their own phone network that i'm integrating this into. Thus the custom default sip uris.**
